pkg/api/types.go

```diff
@@ -1554,7 +1555,7 @@ func (a *AgentPoolProfile) HasImageGallery() bool {
 
 // IsCustomVNET returns true if the customer brought their own VNET
 func (a *AgentPoolProfile) IsCustomVNET() bool {
-	return len(a.VnetSubnetID) > 0
+	return len(a.VnetSubnetID) > 0 && len(a.NodeResourceGroup) > 0 && !strings.Contains(a.VnetSubnetID, a.NodeResourceGroup)
```

Comment:
The comment say we "want to fill VnetSubnetID regardless if the subnet is custom or not", I don't see how this change helps on that aspect? In fact this change is making it stricter to for a subnetId to be considered custom?

Also do you have a link to PR or workitem in RP side? I don't quite understand the intention here. The addition of "NodeResourceGroup" is also a rather big change,  can implement the  logic in RP side (i.e. does not attach to the data type)?

Reply:
I pasted the RP change link to the description.

In the RP pull request I want to fill VnetSubnetID in InheritManagedClusterProperties when vnetsubnetID is empty (which means that customer didn't bring her own vnet subnet)

But then we cannot tell whether customer has custom vnet by whether vnetsubnetID is empty because now it is always not empty.

That's why I need NodeResourceGroup property because the vnet is a custom vnet if it is not in the NodeResourceGroup

Reply:
"the vnet is a custom vnet if it is not in the NodeResourceGroup"  this is not correct as we support BYO node resource group so cx could do both BYO VNET and BYO node ResourceGroup

Reply:
got you and thanks for bringing up this point. Let me examine the BYO node resource group logic to see if it is feasible at all to tell auto-created subnet and BYO subnet apart if we fill VnetSubnetID all the time

Reply:
We need to check if the generated vnet will always be in noderesourcegroup. But it's also always generated with a specific name. If it has that name then we're pretty safe. Wei lets see if we can remember to talk about this on wednesday/thursday.